.github/workflows/build-test-windows.yml

```diff
@@ -0,0 +1,33 @@
+
+name: build-and-test-windows
+on:
+  push:
+    branches: [ main ]
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
+  pull_request:
+
+jobs:
+  windows-unittest:
+    runs-on: windows-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ~1.15.15
+
+      - name: Cache Go
+        uses: actions/cache@v2
+        with:
+          path: |
+            %LocalAppData%\go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Run Unit tests
+        run: go test ./...
```

Comment:
I feel like I'm missing something, but why is there no build step for windows? Will there be anything to cache if we just run `go test`? Should this retain some of the flags present in the Makefile for the test command, like `-v -failfast`?

Reply:
I noticed the same thing as you did.  windows took a long time and it wasn't clear why.  I looked at the open telemetry build and they broke out windows into it's ok workflow.  Doing the same here brought down build time, using cache... i have no idea why... it looks identical.

> Should this retain some of the flags present in the Makefile for the test command, like -v -failfast?
-v is verbose, don't think we need verbose unless something is broken

> I feel like I'm missing something, but why is there no build step for windows?
this runs on a windows runner, nothing extra is needed.